core/src/main/java/io/questdb/std/str/Utf8s.java

```diff
@@ -161,18 +161,10 @@ public static boolean equals(@NotNull Utf8String l, @NotNull Utf8String r) {
     }
 
     public static boolean equals(@NotNull Utf8Sequence l, @NotNull Utf8Sequence r) {
-        if (l == r) {
-            return true;
-        }
         if (l.size() != r.size()) {
             return false;
         }
-        for (int index = 0, n = l.size(); index < n; index++) {
-            if (l.byteAt(index) != r.byteAt(index)) {
-                return false;
-            }
-        }
-        return true;
+        return l == r  || l.equalsAssumingSameSize(r);
```

Comment:
nit: do we really need these extra `l == r` branches in this and other methods? I don't recall any situations when we end up comparing the same object in our code.